src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -134,13 +135,14 @@ public static WwwAuthenticateParameters CreateFromWwwAuthenticateHeaderValue(str
         /// Create the authenticate parameters by attempting to call the resource unauthenticated, and analyzing the response.
         /// </summary>
         /// <param name="resourceUri">URI of the resource.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel operation.</param>
         /// <returns>WWW-Authenticate Parameters extracted from response to the un-authenticated call.</returns>
-        public static async Task<WwwAuthenticateParameters> CreateFromResourceResponseAsync(string resourceUri)
+        public static async Task<WwwAuthenticateParameters> CreateFromResourceResponseAsync(string resourceUri, CancellationToken cancellationToken = default)
```

Comment:
+1

Reply:
@bgavrilMS few questions/ideas:
1. Right now it constructs `HttpRequestMessage` and passes it to `HttpClient.SendAsync()`. Should it just call `GetAsync()` instead?
2. Both `HttpClient` and `HttpRequestMessage` implement `IDisposable`. Should both be disposed? Might speed up the freeing of the socket, avoid [socket exhaustion](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests#issues-with-the-original-httpclient-class-available-in-net).

Reply:
1. I don't see why not. 
2. This is a great point. The guidance for SDKs regards to HttpClient is that SDKs should allow calling apps to inject the HttpClient themselves instead of creating it a new one (socket exhaustion) or having a static one (DNS expiration problems). MSAL allows to do this https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-provide-httpclient

So for this method I'd recommend to actually add another optional parameter `IMsalHttpClientFactory`. if people don't add use it, then we can default to MSAL's .. see `CreateDefaultHttpClientFactory`.

Reply:
Obviously the latter is a bit more complex. An simple improvement would be to just use a static HttpClient.

Reply:
I've added 2 overload, once accepting HTTP client and another HTTP client factory. And kept the existing one which doesn't accept any. Removing it would be a braking change.

Let me in this PR just start using a static instance of HTTP client.

I couldn't figure out how to use `CreateDefaultHttpClientFactory()` since it requires an instance of proxy which in turn is an internal interface and I'm not sure how to properly instantiate it.

Another problem is that this helper is `static class` so not really DI container friendly. A solution for that would be converting it to a non-static, keep the existing static API, maybe mark it as obsolete. Looks like that's what has happened with the similar helper in ADAL earlier.

Reply:
Ah, `PlatformProxyFactory` looks like is the way to instantiate `proxy` which will give me the default factory. Hang on, will do.

Reply:
@bgavrilMS please re-review.